### ISSUE TITLE
HTTP Error 405

### ISSUE THREAD

**Issue by dominic01** [kermitt2/grobid#599]
Could be related to Issue #595 

System: AWS Linux 2. Increased memory to 8GB. It is not crashing and we are not processing any documents as well. But after a long idle time, I am seeing error in HTTP Error 405 in the browser.
PFA: 
<img width="391" alt="Untitled" src="https://user-images.githubusercontent.com/6888834/85654477-f4ce5280-b6cb-11ea-931e-c6cdeb0cf946.png">

But when I try the same from another server using curl, its working. This is really strange.
```
$ curl -X POST -d "names=Simmons Jr. GM" http://10.235.25.203:8070/api/processHeaderNames

<persName xmlns="http://www.tei-c.org/ns/1.0"><forename type="first">Simmons</forename><surname>Gm</surname><genName>Jr</genName></persName>
```

The run log for this period.  
[gro.log](https://github.com/kermitt2/grobid/files/4829136/gro.log)

**Comment by kermitt2** [comment]
It sound a bit like a CORS problem or an issue with the authorized header parameters. You could compare the curl header (`curl -v`) from the aws server with the ones of your browser request. But this behavior points to an issue with domain authorization I think.

**Comment by dominic01** [comment]
Both the headres looks the same. 
The strange thing is, it is not happening immediately after starting Grobid service. We tried few things and leave it idel for the whole night or long period of time,  we are seeing this error. 
Also log is getting updated when no process is running at the client side. Is it due to the "DEBUG" option?

**Comment by kermitt2** [comment]
The headers should be different - very different actually!
Are you using a load balancer, a reverse proxy or any intermediary process/firewall between the public facing access and the instance access? 
I don't have experience with AWS ec2, but the address 22.222.22.222 looks weird I must say.

**Comment by elonzh** [comment]
I think it is not a CORS issue, the path of request from the web page sometimes is `/`.

```
172.17.0.1 - - [29/Jun/2021:08:47:54 +0000] "POST / HTTP/1.1" 405 304 "http://127.0.0.1:8070/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36 Edg/91.0.86
4.59" 12
172.17.0.1 - - [29/Jun/2021:08:47:54 +0000] "POST / HTTP/1.1" 405 304 "http://127.0.0.1:8070/" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36 Edg/91.0.86
4.59" 1
```

**Comment by elonzh** [comment]
After some debug, I found the cause:

![image](https://user-images.githubusercontent.com/8749411/123769390-d8c34980-d8fb-11eb-96e1-7627fd42ec6e.png)

1. The default URL for Grobid web page is `http://127.0.0.1:8070/#`.
2. The form will use `action` to get the URL, for example `$('#gbdForm2').attr('action')` will get "http://127.0.0.1:8070/#api/referenceAnnotations" and it's a wrong url.

**Comment by kermitt2** [comment]
> 1. The default URL for Grobid web page is `http://127.0.0.1:8070/#`.

In principle, all the `<a href="#">` click events are captured by th web app, so we should not arrive to `http://22.222.222:8070/#` as base url. Do you know which link leads to the `/#` ?

**Comment by elonzh** [comment]
Open any tab as a new browser tab.

> > 1. The default URL for Grobid web page is `http://127.0.0.1:8070/#`.
> 
> In principle, all the `<a href="#">` click events are captured by the web app, so we should not arrive to `http://22.222.222:8070/#` as base url. Do you know which link leads to the `/#` ?

**Comment by kermitt2** [comment]
> Open any tab as a new browser tab.

Awww it was thought as a single window application. Damn users :D

**Comment by elonzh** [comment]
Actually, It's a common use case, I almost always open multiple tabs for tests.

**Comment by kermitt2** [comment]
> Actually, It's a common use case, I almost always open multiple tabs for tests.

Yes indeed! it's funny sometimes how we could expect so wrongly/lazily that all the users will follow the same behaviour as ourself. 

Normally it's fixed with b372bf4217758be3bc625a8fd82071ebc7af7fde

Thanks a lot for the brilliant insightful debug !

**Comment by lfoppiano** [comment]
This seems to have been fixed some time ago 😄
